hazelcast/src/main/java/com/hazelcast/spi/impl/sequence/AbstractCallIdSequence.java

```diff
@@ -77,7 +81,13 @@ public void complete() {
     }
 
     public long forceNext() {
-        return longs.incrementAndGet(INDEX_HEAD);
+        long l = longs.incrementAndGet(INDEX_HEAD);
+        // we don't want to check for every call, so we'll check 1 in 8 calls. If there is concurrencyDetection,
+        // &&
+        if (((l >> 3) << 3) == l && concurrentInvocations() > 10) {
```

Comment:
minor: you can use `QuickMath.modPowerOfTwo(l, 8) == 0` instead of `((l >> 3) << 3) == l`, to make it more readable. I misread this like `((l >> 3) << 3) == 1` initially (so `1` instead of small `L`) and didn't get what it means.

Reply:
Thanks I'll use that.